### PR TITLE
feat: add delete button in plugin drawer

### LINKED ISSUE #1279: 当Route中存在被禁用的插件时Service里的同名插件失效
# Bug report

## Describe the bug

当Route中存在被禁用的插件时Service里的同名插件失效。

## How to Reproduce

1. 创建一个Service SRV-A，启用basic-auth插件
2. 创建一个Route ROUTE-A，关联上第一部的SRV-A，启用basic-auth插件
3. 创建一个Consumer，配置basic-auth的账号和密码
4. 编辑ROUTE-A，将其之前启用的basic-auth插件禁用
5. 根据ROUTE-A访问应用时basic-auth失效，因为etcd中对应route配置有{"plugins": {"basic-auth":{"disable": true}}}的配置，且Route里的同名插件basic-auth优先级高于其对应的Service插件。
6. 手工编辑ROUTE-A在etcd里的配置，将其“basic-auth”删除（不仅是disable），再次据ROUTE-A访问应用则basic-auth生效

## Expected behavior

根据ROUTE-A访问应用时basic-auth生效

## Screenshots

Add screenshots to help explain your problem if applicable.

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. Chrome, Safari, Edge]
- Version: [e.g. 2.0-rc3]

## Additional context

建议在配置插件时不仅仅将其enable或者disable，还需要提供**删除插件**的功能。

### PR DESCRIPTION
**Pull request by guoqqqi**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
resolve #1279 
___
### New feature or improvement
- Describe the details and related test reports.
Added delete plugin button

image

### CODE REVIEW

**Comment by guoqqqi** [comment]
Still need to add a delete button method to achieve the function, I will update after～

File: web/src/components/Plugin/PluginDetail.tsx [hunk 1]
```diff
@@ -35,8 +35,10 @@ import { LinkOutlined } from '@ant-design/icons';
 import Ajv from 'ajv';
 import type { DefinedError } from 'ajv';
 import addFormats from 'ajv-formats';
+import { omit } from 'lodash';
 
 import { fetchSchema } from './service';
+import { createOrUpdate } from '@/pages/Plugin/service';
```
Comment (juzhiyuan) [inline web/src/components/Plugin/PluginDetail.tsx:41]: Line41 contains `@/xx`, it should be ahead of Line40.

**Comment by juzhiyuan** [comment]
@guoqqqi LGTM, and please add test cases for this PR.

File: web/src/components/Plugin/PluginDetail.tsx [hunk 2]
```diff
@@ -174,24 +178,49 @@ const PluginDetail: React.FC<Props> = ({
...
-                } catch (error) {
-                  notification.error({
-                    message: 'Invalid JSON data',
+            <Space>
+              <Popconfirm
+                title={formatMessage({ id: 'page.plugin.drawer.popconfirm.title.delete' })}
+                okText={formatMessage({ id: 'component.global.confirm' })}
+                cancelText={formatMessage({ id: 'component.global.cancel' })}
+                onConfirm={() => {
+                  const plugins = omit(initialData, [`${name}`]);
```
Comment (juzhiyuan) [inline web/src/components/Plugin/PluginDetail.tsx:187]:
```suggestion
                  const plugins = omit(initialData, name);
```

File: web/src/pages/Plugin/locales/en-US.ts [hunk 3]
```diff
@@ -14,4 +14,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-export default {};
+export default {
+  'page.plugin.drawer.popconfirm.title.delete': 'Are you sure you want to delete all data from this plugin?',
```
Comment (juzhiyuan) [inline web/src/pages/Plugin/locales/en-US.ts:18]:
```suggestion
  'page.plugin.drawer.popconfirm.title.delete': 'Are you sure to delete this item?',
```

File: web/src/pages/Plugin/locales/zh-CN.ts [hunk 4]
```diff
@@ -14,4 +14,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-export default {};
+export default {
+  'page.plugin.drawer.popconfirm.title.delete': '您确定删除该插件所有数据吗？',
```
Comment (juzhiyuan) [inline web/src/pages/Plugin/locales/zh-CN.ts:18]:
```suggestion
  'page.plugin.drawer.popconfirm.title.delete': '确定删除该插件吗？',
```

**Comment by juzhiyuan** [comment]
@guoqqqi You need to merge the master branch into this one to fix Backend E2E failed.

**Comment by guoqqqi** [comment]
> @guoqqqi You need to merge the master branch into this one to fix Backend E2E failed.

I've updated the branch to the latest, but the CI didn't pass

**Comment by juzhiyuan** [comment]
all right, wait for other PRs to fix this issue.

**Review by juzhiyuan — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Pull request merged**